src/routing-walker.ts

```diff
@@ -39,37 +37,23 @@ export const walkRouting = ({
     );
     const path = `${parentPath || ""}${segment ? `/${segment}` : ""}`;
     if (element instanceof AbstractEndpoint) {
-      const methods: (Method | AuxMethod)[] = element.getMethods().slice();
-      if (hasCors) methods.push("options");
+      const methods = element.getMethods() || ["get"];
       for (const method of methods) onEndpoint(element, path, method);
     } else if (element instanceof ServeStatic) {
       if (onStatic) element.apply(path, onStatic);
     } else if (element instanceof DependsOnMethod) {
       for (const [method, endpoint] of element.pairs) {
+        const supportedMethods = endpoint.getMethods();
         assert(
-          endpoint.getMethods().includes(method),
+          !supportedMethods || supportedMethods.includes(method),
           new RoutingError(
             `Endpoint assigned to ${method} method of ${path} must support ${method} method.`,
           ),
         );
-        onEndpoint(endpoint, path, method);
-      }
-      if (hasCors && element.firstEndpoint) {
-        onEndpoint(
-          element.firstEndpoint,
-          path,
-          "options",
-          element.siblingMethods,
-        );
+        onEndpoint(endpoint, path, method, element.siblingMethods);
```

Comment:
`siblingMethods` implementation did work correct with `firstEndpoint` and needs an adjustment to take all except `method` into account